### PR TITLE
Create xml coverage

### PR DESCRIPTION
**Pull request by steffen-weber**
Add coverage badged from coveralls.io

### CODE REVIEW

**Comment by maximilianhuber** [comment]
I don't think that `src/composer.lock` should be added.

I have restarted the travis tests, since some of them died in the `apt-get`-phase

File: src/composer.json [hunk 1]
```diff
@@ -32,5 +35,11 @@
             "Fossology\\Decider\\UI\\": "decider/ui",
             "Fossology\\UI\\Page\\": "www/ui/page"
         }
+    },
+    
+    "config": {
+        "platform": {
+            "php": "5.3.3"
```
Comment (maximilianhuber) [inline src/composer.json]: Why do you have added this part?

File: .travis.yml [hunk 2]
```diff
@@ -86,3 +86,8 @@ script:
  - if [[ ${TRAVIS_PHP_VERSION:0:3} == "5.5" ]]; then composer require --dev --no-update phpunit/phpunit ~4; fi
  - phpunit -csrc/phpunit.xml --testsuite="$PHPTESTSUITE"
  - set +e
+
+after_script:
+ - mkdir build
+ - mkdir build/logs
```
Comment (maximilianhuber) [inline .travis.yml]: maybe replace with `mkdir -p build/logs`?

**Comment by maximilianhuber** [comment]
Code looks good to me

File: src/composer.json (hunk 1 above)
Comment (steffen-weber) [inline src/composer.json]: Without this configuration, composer update would install package versions which require php>=5.4
We should answer question https://github.com/fossology/fossology/issues/585 soon.

Comment (maximilianhuber) [inline src/composer.json]: I agree with you and I would suggest to drop 5.3 now. We also have released 3.0.0 for 5.3 as suggested in the issue #585 and the next one will be with new features.

@mcjaeger: what is your opinion?

Comment (maximilianhuber) [inline src/composer.json]: Let us postpone this question, I have created a note in #585.

**Comment by maximilianhuber** [comment]
The merge lead to failing travis tests. Will try to fix them.

#### Note:
* All tests where green before the merge. After restarting them some become red.

* The tests fail with the following output (also see screenshot):
```
[...]
  - Installing symfony/console (v2.6.13)
    Downloading: Connecting...    Failed to download symfony/console from dist: Could not authenticate against github.com
    Now trying to download from source
  - Installing symfony/console (v2.6.13)
    Cloning 0e5e18ae09d3f5c06367759be940e9ed3f568359

  - Installing symfony/config (v2.6.13)
    Downloading: Connecting...    Failed to download symfony/config from dist: Could not authenticate against github.com
    Now trying to download from source
  - Installing symfony/config (v2.6.13)
    Cloning b5063937fab6fdfb7bacc00bc8c0cd7ee0c50070
    b5063937fab6fdfb7bacc00bc8c0cd7ee0c50070 is gone (history was rewritten?)


  [RuntimeException]                                                           
  Failed to execute git checkout 'b5063937fab6fdfb7bacc00bc8c0cd7ee0c50070' -  
  - && git reset --hard 'b5063937fab6fdfb7bacc00bc8c0cd7ee0c50070' --          
  fatal: reference is not a tree: b5063937fab6fdfb7bacc00bc8c0cd7ee0c50070                                                                                 
```
although the referenced commit [symfony/config/b5063937fab6fdfb7bacc00bc8c0cd7ee0c50070](https://github.com/symfony/config/commit/b5063937fab6fdfb7bacc00bc8c0cd7ee0c50070) exists.

* Other tests, which only differ in the version of *g++* do not fail

--
tmp

**Comment by maximilianhuber** [comment]
Sometimes it works:
tmp

I have restarted the travis jobs since one of them worked now

**Comment by maximilianhuber** [comment]
The failing test is is [#647](https://travis-ci.org/fossology/fossology/builds/116097694).

The  failing jobs (even after multiple restarts):
* [#647.3](https://travis-ci.org/fossology/fossology/jobs/116097697)
* [#647.12](https://travis-ci.org/fossology/fossology/jobs/116097708)
* [#647.15](https://travis-ci.org/fossology/fossology/jobs/116097711)

The job [#647.5](https://travis-ci.org/fossology/fossology/jobs/116097699) has failed previously but has recovered after restarting it.

**Comment by maximilianhuber** [comment]
@steffen-weber: also other travis tests are failing, which not yet include the changes introduced in this PR. For example [this test](https://travis-ci.org/siemens/fossology/builds/116130804) of a branch on a fork.

I seams that this is a general infrastructure problem.

**Comment by steffen-weber** [comment]
@maximilianhuber  It could be caused by a github limit:
https://getcomposer.org/doc/articles/troubleshooting.md#api-rate-limit-and-oauth-tokens
With `composer install --prefer-source` as in [this commit](https://github.com/steffen-weber/fossology/commit/1cdf0805c869eb8adc5c941a90475ccb30aa811c), the test set-up was slower (by factor 4) but 
after failing to execute git checkout travis successfully downloaded from dist.

**Comment by maximilianhuber** [comment]
Am I right that the parameter `--prefer-source` only changes the fallback order from
```
    dist > source
```
to 
```
    source > dist
```
and that this makes it slow enough that it don't reaches the api-rate-limit?

Your commit does not look like a clean solution.

**Comment by steffen-weber** [comment]
It might be a dirty solution and I'm not even sure if it will fix the trouble. It was just an idea for further investigations.

**Comment by maximilianhuber** [comment]
> although the referenced commit symfony/config/b5063937fab6fdfb7bacc00bc8c0cd7ee0c50070 exists.

They have rebased their history. The correct hash for this commit is now [0ca496cbe208fc37c4cf3415ebb3056e0963115b](https://github.com/symfony/config/commit/0ca496cbe208fc37c4cf3415ebb3056e0963115b). We should change that in the `composer.lock` file.

BUT: this does not solve the problem, that composer falls back to source download.

This does not only occur in travis, also an local machines which live behind a proxy the `composer install` step has the same problem.

**Pull request merged**